src/Fixer/ControlStructure/NoBreakCommentFixer.php

```diff
@@ -0,0 +1,266 @@
+                new CodeSample("<?php\nswitch (\$foo)\n{    case 1:\n        foo();\n        // no break\n        break;\n    case 2:\n        bar();\n}"),
+            ),
+            'Adds a "no break" comment before fall-through cases, and removes it if there is no fall-through.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(array(T_CASE, T_DEFAULT));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $position => $token) {
+            if ($token->isGivenKind(array(T_CASE, T_DEFAULT))) {
+                $this->fixCase($tokens, $position);
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $casePosition
+     */
+    private function fixCase(Tokens $tokens, $casePosition)
+    {
+        $empty = true;
+        $fallThrough = true;
+        $commentPosition = null;
+        for ($i = $tokens->getNextTokenOfKind($casePosition, array(':', ';')) + 1, $max = count($tokens); $i < $max; ++$i) {
+            if ($tokens[$i]->isGivenKind(array(T_SWITCH, T_IF, T_ELSE, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_DO, T_FUNCTION, T_CLASS))) {
+                $empty = false;
+                $i = $this->getStructureEnd($tokens, $i);
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_BREAK, T_CONTINUE, T_RETURN))) {
+                $fallThrough = false;
+
+                continue;
+            }
+
+            if ($tokens[$i]->equals('}')) {
+                if (null !== $commentPosition) {
+                    $this->removeComment($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if ($this->isNoBreakComment($tokens[$i])) {
+                $commentPosition = $i;
+
+                continue;
+            }
+
+            if ($tokens[$i]->isGivenKind(array(T_CASE, T_DEFAULT))) {
+                if ($empty || !$fallThrough) {
+                    if (null !== $commentPosition) {
+                        $this->removeComment($tokens, $commentPosition);
+                    }
+                } elseif (null === $commentPosition) {
+                    $this->insertCommentAt($tokens, $i);
+                } else {
+                    $this->ensureNewLineAt($tokens, $commentPosition);
+                }
+
+                break;
+            }
+
+            if (!$tokens[$i]->isGivenKind(array(T_COMMENT, T_WHITESPACE))) {
+                $empty = false;
+            }
+        }
+    }
+
+    /**
+     * @param Token $token
+     *
+     * @return bool
+     */
+    private function isNoBreakComment(Token $token)
+    {
+        if (!$token->isGivenKind(array(T_COMMENT, T_DOC_COMMENT))) {
```

Comment:
please use the short hand [!$token->isComment()](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.2/src/Tokenizer/Token.php#L352)